File: examples/multistep_form.py
"""
Multi-step Form Example

This example demonstrates a multi-step form with progress tracking.
"""

from formora import Form, CssFramework, Rule

html = (
    Form("onboarding")
    .title("Welcome! Let's get you set up.")
    .description("This will take about 2 minutes.")
    .css(CssFramework.TAILWIND)
    # Step 1: Basic Info
    .step("Basic Information")
    .text("full_name", "Full Name", required=True, rules=[Rule.required()])
    .email("email", "Email Address", required=True, rules=[Rule.required(), Rule.email()])
    .text("company", "Company Name")
    # Step 2: Role Details
    .step("Your Role")
    .select("role", "What's your role?", options=[
        ("Developer", "developer"),
        ("Designer", "designer"),
        ("Manager", "manager"),
        ("Other", "other")
    ], required=True)
    .number("years_experience", "Years of Experience", min=0, max=50)
    .multi_select("technologies", "Technologies (select all that apply)", options=[
        ("Python", "python"),
        ("JavaScript", "javascript"),
        ("Rust", "rust"),
        ("Go", "go"),
        ("TypeScript", "typescript")
    ])
    # Step 3: Preferences
    .step("Preferences")
    .select("frequency", "How often do you want to hear from us?", options=[
        ("Daily", "daily"),
        ("Weekly", "weekly"),
        ("Monthly", "monthly")
    ])
    .checkbox("terms", "I agree to the terms and conditions", default=True)
    .checkbox("privacy", "I agree to the privacy policy", default=True)
    .submit_label("Complete Setup")
    .success_message("Setup complete! Welcome aboard.")
    .build()
)

print(html)

File: formora-js/examples/multistep_form.ts
/**
 * Multi-step onboarding form example.
 * TypeScript equivalent of examples/multistep_form.py
 */

import init, { Form, Rule, CssFramework } from "../wasm/formora_js";

async function main() {
  await init();

  const html = new Form("onboarding")
    .title("Welcome to Formora")
    .description("Complete your profile in 3 easy steps.")
    .css(CssFramework.tailwind())
    // Step 1 — Personal Info
    .step("Personal Information")
    .text("full_name", "Full Name", "Your full name", true, undefined, undefined, [
      Rule.required(),
      Rule.minLength(2),
    ])
    .email("email", "Work Email", true, undefined, undefined, [
      Rule.required(),
      Rule.email(),
    ])
    .text("company", "Company", "Your company name", true)
    // Step 2 — Role
    .step("Role & Experience")
    .select(
      "role",
      "Your Role",
      [
        ["Developer", "developer"],
        ["Designer", "designer"],
        ["Product Manager", "pm"],
        ["Data Scientist", "data_scientist"],
        ["Other", "other"],
      ],
      true
    )
    .range("years_exp", "Years of Experience", 0, 30, 1, 0)
    .multiSelect(
      "technologies",
      "Technologies You Use",
      [
        ["TypeScript", "typescript"],
        ["Rust", "rust"],
        ["Python", "python"],
        ["React", "react"],
        ["Node.js", "nodejs"],
      ]
    )
    // Step 3 — Preferences
    .step("Preferences")
    .radio(
      "frequency",
      "How often do you ship?",
      [
        ["Daily", "daily"],
        ["Weekly", "weekly"],
        ["Monthly", "monthly"],
      ],
      true
    )
    .checkbox("agree_terms", "I agree to the Terms of Service", false, undefined, [
      Rule.required("You must accept the terms"),
    ])
    .checkbox("agree_privacy", "I agree to the Privacy Policy", false, undefined, [
      Rule.required("You must accept the privacy policy"),
    ])
    .submitLabel("Complete Setup")
    .successMessage("You're all set! Welcome aboard.")
    .build();

  console.log(html);
}

main().catch(console.error);
